virtool_core/models/reference.py

```diff
@@ -18,13 +18,26 @@ class ReferenceDataType(str, enum.Enum):
     genome = "genome"
 
 
-class ReferenceUser(UserNested):
+class ReferenceRights(BaseModel):
     build: bool
     modify: bool
     modify_otu: bool
     remove: bool
 
 
+class ReferenceGroupOrUser(ReferenceRights):
```

Comment:
Please split into separate models the inherit from `ReferenceRights`. If they diverge at some point (they will), a refactor less painful.
```suggestion
class ReferenceGroupOrUser(ReferenceRights):
```